Mathlib/GroupTheory/GroupAction/Defs.lean

```diff
@@ -661,6 +661,13 @@ theorem isPretransitive_compHom
   obtain ⟨e, rfl⟩ : ∃ e, f e = m := hf m
   exact ⟨e, rfl⟩
 
+@[to_additive]
+theorem isPretransitive_of_compHom
```

Comment:
This name will make it possible to use new-style dot notation:
```suggestion
theorem IsPretransitive.of_compHom
```
Also, I think that we should have `MulActionHom.isPretransitive` instead of this lemma.

Reply:
Ah, sure, that makes sense. I'll look into this when I get the chance.

Reply:
`MulActionHom` is for when one group acts on two sets, and you have an equivariant map between the sets. But here we have two groups acting on one set, and a group homomorphism between the groups.

Reply:
You're right but I think that we should have a version that doesn't assume definitional equalities anyway. Something like (not tested)
```
theorem IsPretransitive.of_smul_eq {M N α : Type*} [SMul M α] [SMul N α]
    [IsPretransitive M α] (f : M → N) (hf : ∀ {c : M} {x : α}, f c • x = c • x) :
    IsPretransitive N α :=
```
with corollaries for `of_compHom` and `IsScalarTower` (do we already have it?).
And while you're on it, please add versions about `IsTransitive`.

Also, please use `M`, `N` for monoids and `α` for a type without any structure.

Reply:
Thanks! I've added this version you suggested. I'm not sure what the `IsScalarTower` version would be (since an `IsScalarTower M N α` doesn't come with a function `f : M → N`). And I thought that mathlib didn't have `IsTransitive`, only `IsPretransitive`.

Reply:
You're right about `IsTransitive`. About `IsScalarTower`: in this case `f x = x • 1` and the required equality is provided by [smul_one_smul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/GroupAction/Defs.html#smul_one_smul).

Reply:
Ah, of course. Thanks!